test/test_train_mp_imagenet.py

```diff
@@ -197,32 +197,39 @@ def train_loop_fn(loader):
       tracker.add(FLAGS.batch_size)
       if lr_scheduler:
         lr_scheduler.step()
-      if x % FLAGS.log_steps == 0:
-        xm.add_step_closure(_train_update, args=(device, x, loss, tracker))
+      if step % FLAGS.log_steps == 0:
+        step = '{}/{}'.format(step, len(train_loader))
```

Comment:
Please do not add code to this function.
Any code added here is perceived as clutter, to someone expecting normal pytorch code.
Move the extra code in the closure.
Also the `len()` on a loader can be implemented badly, and having code that pokes at it within the loop might make things worse, perf wise.

Reply:
I feel there is value in seeing the total number of steps, but I can delete the `len` if you don't feel so.